docs/pages/database-access/guides/azure-sql-server-ad.mdx

```diff
@@ -99,77 +100,152 @@ The newly created user will be attached to the public role, which might not
+
+### Create a custom role
+
+Teleport requires `<resource-type>/read` permissions for discovery, but
+only for the resource types you have.
+
+Here is a sample role definition allowing Teleport to read **Azure SQL Servers**
+and **Azure SQL Managed Instances**:
+
+```json
+{
+    "properties": {
+        "roleName": "SQLServerAutoDiscovery",
+        "description": "Allows Teleport to discover SQL Servers and SQL Managed Instances.",
+        "assignableScopes": [
+            "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        ],
+        "permissions": [
+            {
+                "actions": [
+                    "Microsoft.Sql/managedInstances/read",
+                    "Microsoft.Sql/servers/read"
+                ],
+                "notActions": [],
+                "dataActions": [],
+                "notDataActions": []
+            }
+        ]
+    }
+}
+```
+
+The `assignableScopes` field above includes a subscription
+`/subscriptions/<subscription>`, allowing the role to be assigned at any
+resource scope within that subscription or the subscription scope itself.
+If you want to further limit the `assignableScopes`, you can use a resource
+group (`/subscriptions/<subscription>/resourceGroups/<group>`) or a management
+group (`/providers/Microsoft.Management/managementGroups/<group>`) instead.
+
+Now go to the [Subscriptions](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBlade)
+page and select a subscription.
+
+Click on **Access control (IAM)** in the subscription and select **Add** >
+**Add custom role**:
+
+![IAM custom role](../../../img/database-access/guides/azure/add-custom-role@2x.png)
+
+In the custom role creation page, click the **JSON** tab and click **Edit**,
+then paste the JSON example and replace the subscription in `assignableScopes`
+with your own subscription id:
+
+![Create JSON role](../../../img/database-access/guides/sqlserver/create-role-from-json.png)
+
+### Create a role assignment for the Teleport Database Service principal
+
+(!docs/pages/includes/database-access/azure-assign-service-principal.mdx!)
+
+## Step 8/9. Start Teleport Database Service
+
+Start the Database Service:
+
+```code
+$ teleport start --config=/etc/teleport.yaml
+```
+
 <Admonition
   type="tip"
   title="Tip"
 >
   A single Teleport process can run multiple different services, for example
   multiple Database Service agents as well as the SSH Service or Application
   Service.
-  
 </Admonition>
 
-Start the Database Service:
-
-```code
-$ teleport start --config=/etc/teleport.yaml
-```
-
-## Step 7/7. Connect
+## Step 9/9. Connect
 
 Log in to your Teleport cluster. Your database should appear in the list of
 available databases:
 
 <ScopedBlock scope={["oss", "enterprise"]}>
+
 ```code
 $ tsh login --proxy=teleport.example.com --user=alice
 $ tsh db ls
 # Name      Description         Labels
 # --------- ------------------- -------
 # sqlserver                     env=dev
 ```
```

Comment:
Could we give an example (or examples) of the discovered SQL server with populated descriptions and labels?